File: TenMinsJava/day89.java
// Ask for a text, search email or name then output matches
package TenMinsJava;

import java.io.File;
import java.util.ArrayList;
import java.util.Scanner;

public class day89 {

    public static String[] searchMultiple(String name) {

        ArrayList<String> result = new ArrayList<>();
        
        try {
            File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
            Scanner fScan = new Scanner(file);
            
            while (fScan.hasNextLine()) {
                String line = fScan.nextLine().trim();
                String temp [] = line.split(",");
                for (String s : temp) {
                    if (s.matches(name)) {
                        result.add(line);
                    }
                }
            }

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }

        // Convert ArrayList<String> → String[]
        String[] output = new String[result.size()];
        for (int i = 0; i < result.size(); i++) {
            output[i] = result.get(i);
        }

        return output;
    }
}

File: TenMinsPy/day89.py
# Ask for a text, search email or name then output matches

import csv

def search_multiple(name, file_path):
    result = []

    try:
        with open(file_path, newline='', encoding='utf-8') as csvfile:
            reader = csv.reader(csvfile)
            for row in reader:
                # Check each column for exact match
                for cell in row:
                    if name in cell:
                        result.append(','.join(row))  # Join row as string

    except Exception as e:
        print(f"Error: {e}")

    return result